my_init.d/mount-ssh-from-secret.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash -eux
+
+if [ ! -d /home/${SYSTEM_USER}/.ssh/ ] && [ -f /run/secrets/ssh_key ]; then
+  mkdir --mode=0700 /home/${SYSTEM_USER}/.ssh/
+  touch /home/${SYSTEM_USER}/.ssh/known_hosts
+  # Linking ssh key from secret.
+  cp /run/secrets/ssh_key /home/${SYSTEM_USER}/.ssh/id_rsa
+
+  # Try to fix permissions (should fail when user was not yet created).
+  chown -R ${SYSTEM_USER}:${SYSTEM_USER} /home/${SYSTEM_USER}/.ssh || echo "initial startup"
```

Comment:
I don't quite understand the `||` fallback here - shouldn't it say something like "Failed to change permissions - user ${SYSTEM_USER} may not yet exist" ?

If the `chown` fails, then the .ssh folder will have been created, and according to your logic the startup script will no longer do anything.

Reply:
By the way, depending on Dave's use case, this could be made more aggressive if we choose a non-standard name for the SSH key (e.g. `docker_secret`).

The risk of having a name collision inside the `.ssh` folder then is basically zero, so you can run the scrip on the key also when the `.ssh` folder already exists.